torch/functional.py

```diff
@@ -278,6 +279,46 @@ def isinf(tensor):
     return tensor.abs() == inf
 
 
+def meshgrid(*tensors, **kwargs):
+    r"""Take :math:`N` tensors, each of which can be either scalar or 1-dimensional
+vector, and create :math:`N` N-dimensional grids, where the :math:`i`th grid is defined by
+expanding the :math:`i`th input over dimensions defined by other inputs.
+
+
+    Args:
+        tensors (list of Tensor): list of scalars or 1 dimensional tensors. Scalars will be
+        treated as tensors of size :math:`(1,)` automatically
+
+    Returns:
+        seq (sequence of Tensors): If the input has :math:`k` tensors of size
+        :math:`(N_1,), (N_2,), \ldots , (N_k,)`, then the output would also has :math:`k` tensors,
+        where all tensors are of size :math:`(N_1, N_2, \ldots , N_k)`.
+
+    Example::
+
+        >>> x = torch.tensor([1, 2, 3])
+        >>> y = torch.tensor([4, 5, 6])
+        >>> grid_x, grid_y = torch.meshgrid(x, y)
+        >>> grid_x
+        tensor([[1, 1, 1],
+                [2, 2, 2],
+                [3, 3, 3]])
+        >>> grid_y
+        tensor([[4, 5, 6],
+                [4, 5, 6],
+                [4, 5, 6]])
+    """
+    if kwargs:
+        raise TypeError("meshgrid() got an unexpected keyword argument '%s'" % (list(kwargs)[0],))
+    if len(tensors) == 1 and isinstance(tensors[0], (list, tuple)):
+        # the old interface of passing the operands as one list argument
+        tensors = tensors[0]
+    for tensor in tensors:
```

Comment:
If some of these  is not a Tensor, our cpp arg parse will catch it, I think, and will give a more meaningful error message than this.

Reply:
#10823 had a requirement that torch.meshgrid should assert that inputs are tensors, should I just remove this or do you have suggestions to handle this differently?

Reply:
I think @zou3519 meant in that issue that the tensors should have same (device, dtype) combination. I might be wrong though! cc @zou3519

Reply:
Yes, @SsnL is right. I think if you remove it everything will be okay (check the error message though, please :) ); checking that the tensors should have the same (device, dtype) combination should happen in the C++ backend (feel free to do this in a followup PR or this one, @upriser7)

Reply:
I will remove this and add the tensor checks in the C++ backend shortly, thank you for the help.